### PR TITLE
feat: discovery starting point model

### PR DESCRIPTION
**Pull request by xnought**
~Don't merge this yet, still refactoring and fixing.~ This is mergable now, check the comment below for more details

### CODE REVIEW

**Comment by xnought** [comment]
Screen Shot 2022-10-10 at 6 18 26 PM

# Summary
- Starting points show a scatterplot thumbnail of the overall scatter you are filtering from
- User can lasso a new starting point (saved only as ids)
- Users filter + project down from a given starting point

# Note
1. The filter + project mechanism is much better when you can view 2 scatterplots (starting and current)
2. BUT, its somewhat unintuitive for how the starting points relates to the left sidebar.
3. For clip, filter + project does absolutely nothing and not helpful

This was my hypothesis a couple months ago too. Perhaps the uncertainty in the embedding directly from the neural network that was trained is helpful for filter+reproject, but a default embedding (like clip) does not reveal anything new. Although clip is better at the start and for global exploration (not filter).

I'm struggling to justify filter + reproject. It was not helpful at all in my 10+ hour exploration of imagenet validation. I'm going to phase it out if no one can come up with a good reason why it should be used. + not interactive at all when running reprojection.

File: frontend/src/discover/scatter/AutoScaleRegl.svelte [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
-	import type { ScatterRows, ReglConfig, ColorRange } from "./scatter";
+	import type { ScatterRowsFormat, ReglConfig, ColorRange } from "./scatter";
 
 	export let width: number;
 	export let height: number;
-	export let data: ScatterRows = [];
+	export let data: ScatterRowsFormat = [];
 	export let colorRange: ColorRange = [];
 	export let config: ReglConfig = {};
+	export let resetSelection = true;
```
Comment (cabreraalex) [inline frontend/src/discover/scatter/AutoScaleRegl.svelte:12]: remove, not being used?

File: frontend/src/discover/scatter/Scatter.svelte [hunk 2]
```diff
@@ -1,38 +1,54 @@
...
 	export let config: ReglConfig = {};
+	export let resetSelection = true;
+	export let inferDimensions = false;
 
-	let innerWidth, innerHeight;
-	let width;
-	let height = 500;
-	$: width = innerWidth - 470;
+	let innerWidth: number, innerHeight: number;
+	export let width = 500;
```
Comment (cabreraalex) [inline frontend/src/discover/scatter/Scatter.svelte:15]: move above with exports

File: frontend/src/metadata/metadata.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import type ColumnTable from "arquero/dist/types/table/column-table";
 
 import * as aq from "arquero";
 import {
+	interpolateYlGnBu,
 	interpolatePurples,
```
Comment (cabreraalex) [inline frontend/src/metadata/metadata.ts:7]: remove purples import

File: zeno/mirror/sdm.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import json
+import os
+
+import pandas as pd
+
+
+EXAMPLE_SLICES = False
+
+
+def domino(cache_path, pd_with_preds_embed: pd.DataFrame, num_slices: int = 10):
```
Comment (cabreraalex) [inline zeno/mirror/sdm.py:10]: So is this hardcoded currently?

Comment (cabreraalex) [reply zeno/mirror/sdm.py:10]: (totally fine that it is, just curious)

**Review by cabreraalex — CHANGES_REQUESTED** [review]
The points on my screen are super small, not really usable. Can we change the scaling a bit?

File: zeno/mirror/sdm.py (hunk 4 above)
Comment (xnought) [reply zeno/mirror/sdm.py:10]: Yeah I'm running it in a colab notebook to experiment right now. And it actually has a gpu so it doesn't take 10 years on my laptop.

**Comment by xnought** [comment]
What do the points look like for you? For me they are pretty sizable and auto adjust based on the number of points showing.

**Review by cabreraalex — APPROVED** [review]

**Comment by xnought** [comment]
@cabreraalex Okay this is mergable now: I updated the point size and changed the color scale. I can't press the merge button, it's too much power. The ring will corrupt me.
4k_fellowship_movie_screencaps com_5395

**Pull request merged**